### ISSUE TITLE
the losses of auto lane are not converging on CurveLanes Dataset

### ISSUE THREAD

**Issue by TuringKi** [huawei-noah/vega#47]
I test the loss terms  in vega/search_space/networks/pytorch/detectors/auto_lane_detector.py:

 
         `image = input
        loc_targets = kwargs['gt_loc']
        cls_targets = kwargs['gt_cls']

        feat = self.extract_feat(image)
        predict = self.head(feat)

        loc_preds = predict['predict_loc']
        cls_preds = predict['predict_cls']
        cls_targets = cls_targets[..., 1].view(-1)
        pmask = cls_targets > 0
        nmask = ~ pmask
        fpmask = pmask.float()
        fnmask = nmask.float()
        cls_preds = cls_preds.view(-1, cls_preds.shape[-1])
        loc_preds = loc_preds.view(-1, loc_preds.shape[-1])
        loc_targets = loc_targets.view(-1, loc_targets.shape[-1])
        total_postive_num = torch.sum(fpmask)
        total_negative_num = torch.sum(fnmask)  # Number of negative entries to select
        negative_num = torch.clamp(total_postive_num * self.NEGATIVE_RATIO, max=total_negative_num, min=1).int()
        positive_num = torch.clamp(total_postive_num, min=1).int()
        # cls loss begin
        bg_fg_predict = F.log_softmax(cls_preds, dim=-1)
        fg_predict = bg_fg_predict[..., 1]
        bg_predict = bg_fg_predict[..., 0]
        max_hard_pred = find_k_th_small_in_a_tensor(bg_predict[nmask].detach(), negative_num)
        fnmask_ohem = (bg_predict <= max_hard_pred).float() * nmask.float()
        total_cross_pos = -torch.sum(self.ALPHA * fg_predict * fpmask)
        total_cross_neg = -torch.sum(self.ALPHA * bg_predict * fnmask_ohem)
        # class loss end
        # regression loss begin
        length_weighted_mask = torch.ones_like(loc_targets)
        length_weighted_mask[..., self.LANE_POINTS_NUM_DOWN] = 10
        valid_lines_mask = pmask.unsqueeze(-1).expand_as(loc_targets)
        valid_points_mask = (loc_targets != 0)
        unified_mask = length_weighted_mask.float() * valid_lines_mask.float() * valid_points_mask.float()
        smooth_huber = huber_fun(loc_preds - loc_targets) * unified_mask
        loc_smooth_l1_loss = torch.sum(smooth_huber, -1)
        point_num_per_gt_anchor = torch.sum(valid_points_mask.float(), -1).clamp(min=1)
        total_loc = torch.sum(loc_smooth_l1_loss / point_num_per_gt_anchor)
        # regression loss end
        total_cross_pos = total_cross_pos / positive_num
        total_cross_neg = total_cross_neg / positive_num
        total_loc = total_loc / positive_num`

on the CurveLanes Dataset, using the provided optimizer parameters,such as lr=0.02, weight_decay=1e-4,momentum=0.9, etc.
And build a model structrue just like the readme file"https://github.com/huaweinoah/vega/blob/master/docs/en/algorithms/auto_lane.md".

However the loss values are pretty large:
loss_pos = 36.0+
  loss_neg = 10.+
  loss_loc = 100.0+

and did not convergen after 12 epoches, which is mentioned in this paper :https://arxiv.org/abs/2007.12147

Does the loss terms in auto_lane_detector.py are wrong? or just I miss some important steps?

**Comment by sptj** [comment]
Please provide the yml file which config all parameters and more information for the program.
